core/jobs/batch_jobs/audit_rule_validation_jobs.py

```diff
@@ -0,0 +1,684 @@
+                if (
+                    len(ele) > 1
+                ):
+                    invalid_rules.append(rule_spec)
+
+            if (
+                rule_spec.rule_type ==
+                'IsEqualToOrderingWithOneItemAtIncorrectPosition'
+            ):
+                invalid_rules.append(rule_spec)
+
+        return len(invalid_rules) == len(answer_group.rule_specs)
+
+    @staticmethod
+    def invalid_drag_drop_interactions(
+        states_dict: Dict[str, state_domain.State]
+    ) -> List[Dict[str, str]]:
+        """DragAndDropInput interaction contains some invalid rules
+        which we plan to remove. Need to check if all the rules inside
+        the answer group is invalid and after removing it should not
+        result in disconnection of current state to any other state.
+        We will perform this by checking the number of invalid answer
+        groups and the number of answer groups having destination other than
+        try again are equal then we will store the state name and ans group
+
+        Args:
+            states_dict: dict[str, state_domain.State]. The state dictionary.
+            exp_lang_code: str. The language code of the exploration.
+
+        Returns:
+            invalid_states: List[Dict[str, str]]. List of invalid states with
+            the invalid answer groups.
+        """
+        invalid_states = []
+        for state_name, state in states_dict.items():
+            if state.interaction.id != 'DragAndDropSortInput':
+                continue
+            multi_item_value = (
+                state.interaction.customization_args
+                ['allowMultipleItemsInSamePosition'].value
+            )
+            answer_groups = state.interaction.answer_groups
+            invalid_ans_group_count = 0
+            valid_dest_ans_group_count = 0
+            invalid_ans_groups = []
+            for ans_group_idx, answer_group in enumerate(answer_groups):
+                if answer_group.outcome.dest == state_name:
+                    continue
+                valid_dest_ans_group_count += 1
+                can_delete_ans_group = (
+                    ExpAuditRuleChecksJob._filter_invalid_drag_ans_group(
+                        answer_group, multi_item_value)
+                )
+                if can_delete_ans_group:
+                    invalid_ans_group_count += 1
+                    invalid_ans_groups.append(ans_group_idx)
+
+            if (
+                invalid_ans_group_count == valid_dest_ans_group_count and
+                invalid_ans_group_count != 0 and
+                valid_dest_ans_group_count != 0
+            ):
+                invalid_states.append(
+                    {
+                        'state_name': state_name,
+                        "ans_group_idx": invalid_ans_groups
+                    }
+                )
+
+        return invalid_states
+
+    @staticmethod
+    def continue_interac_text_value_language(
```

Comment:
I don't get this function. It either returns an empty list if all continue buttons are valid, or it returns a list with a single item (language code) if it is invalid. It never returns a list with more than a single item. Is this what you wanted? Have you tested this? What's the point of the language codes?

Reply:
Yes, I mean I shouldn't return the list I should simply return the `language_code`. Actually for all the interactions that have text value more than 20 we are planing to set the value to `Continue` but as we know that our explorations are not only in English, so I'm collecting the language code of errored explorations so that we can have `Continue` converted according to the different language.
Thanks @EricZLou

Reply:
Got it. I think the code would be clearer if this function just returned a boolean for whether there is an invalid Continue button. In this function, you're just passing in the `language_code` and returning it back, which doesn't make too much sense at first.

Reply:
I will update that soon, Thanks for the review.